Vp4670 aem

krystian-hebel opened the pull request.

krystian-hebel committed:
tests/aem_hw.pm: reset TPM for vp4670

Signed-off-by: Krystian Hebel <krystian.hebel@3mdeb.com>

krystian-hebel committed:
tests/install_startup.pm: don't hardcode the boot order

Signed-off-by: Krystian Hebel <krystian.hebel@3mdeb.com>

krystian-hebel committed:
tests/aem_hw.pm: update for new ACM package release

Signed-off-by: Krystian Hebel <krystian.hebel@3mdeb.com>

krystian-hebel commented:
Before rebase it worked until TXT error in `aem-first-boot`, the rest shouldn't require changes after TXT stuff is sorted out. Haven't tested if the rebase broke anything.

SergiiDmytruk commented on tests/aem_hw.pm:
```diff
@@ -339,6 +343,47 @@ sub clear_tpm_hp {
+sub clear_tpm_dasharo {
+    # enter setup menu
+    assert_screen 'dasharo_post_delay';
+    send_key 'delete'; # differs between platforms, maybe use OCR if fast enough?
```
I don't remember OCR being slow, so should be a valid option.  Extracting it out of a serial output with a regexp is another one.

krystian-hebel replied:
I'll check what keys are used on all our platforms, maybe they are compatible enough (e.g. no F10 that would show the confirmation) to just spam all the buttons - this is how users are entering the setup menu after all :wink:

SergiiDmytruk commented on tests/aem_hw.pm:
```diff
@@ -393,33 +393,33 @@ sub run_cmd {
+    ### FIXME: handle optiplex
     my $url = 'https://cdrdv2.intel.com/v1/dl/getContent/630744';
-    my $zip_root = '630744_003'; ### FIXME: update to new version, handle optiplex
+    my $zip_root = '630744_004';
```
I guess we can combine old and new archives and store the result locally or on some server.

krystian-hebel replied:
We're discussing this internally. It would be easier if Intel had added a license to these archives...

krystian-hebel replied:
ACMs are now mirrored on https://dl.3mdeb.com/mirror/intel/acm/ and the test will use those to avoid surprises.

SergiiDmytruk commented on tests/aem_hw.pm:
```diff
@@ -393,33 +393,33 @@ sub run_cmd {
+    assert_script_run("cd '/tmp/$zip_root'");
     assert_script_run("unzip -o '$zip_fname'");
-    assert_script_run("cp --update '$zip_root/$bin_fname' /boot");
+    assert_script_run("cp --update *.bin /boot");
```
Copying all ACMs results in tests failure because GRUB waits for user's input after printing more lines that fit on the screen and pager is enabled.  Can either copy only the file we need or run `set pager=0` as a workaround.

krystian-hebel replied:
`set pager=0` is not enough. Because `slaunch_module` for incompatible ACMs returns error, GRUB still asks for confirmation before booting.
![image](https://github.com/user-attachments/assets/159a7d0f-f5b5-4371-8ef9-8e42a81aef95)
I thought we improved it already, maybe it lives in some forgotten branch. Anyway, I'll modify the tests to copy just one ACM as it was done previously, and we can work on GRUB error handling later.

krystian-hebel replied:
Changed in https://github.com/3mdeb/openqa-tests-qubesos/pull/9/commits/e9d8ab5d5566effe11ae384cfcd3e70ba751d3c9, running tests now

SergiiDmytruk replied:
`Press any key to  continue...` shouldn't be a problem unless the entry is run after editing when [GRUB waits indefinitely](https://github.com/TrenchBoot/grub/blob/7259d55ffcf124e32eafb61aa381f9856e98a708/grub-core/normal/menu_entry.c#L1458-L1459), otherwise the boot continues [after a 10s delay](https://github.com/TrenchBoot/grub/blob/7259d55ffcf124e32eafb61aa381f9856e98a708/grub-core/normal/menu.c#L62-L74).  But I do like not having the screen filled with irrelevant errors about lots of ACMs, the only drawback is that this doesn't test handling of multiple Slaunch modules.

krystian-hebel replied:
Darn, it was `O`, not `0` :man_facepalming: https://github.com/3mdeb/openqa-tests-qubesos/compare/764d5c9539442747a5cb9196f9c0983a9423581f..d9799a4881685c4915f1bbc417f7d8b6de5aee83

SergiiDmytruk reviewed (changes requested):
Tests stalls before hitting TXT issue, two other comments are informative.

krystian-hebel committed:
tests/aem_hw.pm: rework setup_acm

- Use mirrored copy of ACMs, Intel archive isn't stable enough
- Download file on SUT directly, no need to scp it
- Copy only one file (avoids GRUB confirmation prompt)
- Add explicit mapping from MACHINE to ACM filename

Signed-off-by: Krystian Hebel <krystian.hebel@3mdeb.com>

krystian-hebel commented:
Now it no longer fails on GRUB, instead it bootloops after TXT starts, see tests 3030 and 3031 on our openQA instance.
